CMakeLists.txt

```diff
@@ -556,28 +558,48 @@ if(FREEBSD OR MACOS)
-                execute_process(COMMAND
-                                brew --prefix --installed openssl
-                                RESULT_VARIABLE BREW_OPENSSL
-                                OUTPUT_VARIABLE BREW_OPENSSL_PREFIX
-                                OUTPUT_STRIP_TRAILING_WHITESPACE)
+if(ENABLE_WOLFSSL)
+        pkg_check_modules(TLS IMPORTED_TARGET REQUIRED wolfssl)
 
-                if((BREW_OPENSSL NOT EQUAL 0) OR (NOT EXISTS "${BREW_OPENSSL_PREFIX}"))
-                        message(FATAL_ERROR "OpenSSL (or LibreSSL) is required for building Netdata, but could not be found.")
-                endif()
+        list(APPEND CMAKE_REQUIRED_LIBRARIES wolfssl)
+        check_function_exists(wolfSSL_set_alpn_protos HAVE_WOLFSSL_SET_ALPN_PROTOS)
+        if(NOT HAVE_WOLFSSL_SET_ALPN_PROTOS)
+            message(FATAL_ERROR "Your WolfSSL library has not been compiled with the OPENSSL_EXTRA flag, which is necessary to create symbols for the OpenSSL API that Netdata uses.")
+        endif()
+else()
+        # openssl/crypto
+        pkg_check_modules(TLS IMPORTED_TARGET openssl)
+
+        if(NOT TARGET PkgConfig::OPENSSL)
+                if(MACOS)
+                        execute_process(COMMAND
+                                        brew --prefix --installed openssl
+                                        RESULT_VARIABLE BREW_OPENSSL
+                                        OUTPUT_VARIABLE BREW_OPENSSL_PREFIX
+                                        OUTPUT_STRIP_TRAILING_WHITESPACE)
+
+                        if((BREW_OPENSSL NOT EQUAL 0) OR (NOT EXISTS "${BREW_OPENSSL_PREFIX}"))
+                                message(FATAL_ERROR "OpenSSL (or LibreSSL) is required for building Netdata, but could not be found.")
+                        endif()
 
-                set(OPENSSL_INCLUDE_DIRS "${BREW_OPENSSL_PREFIX}/include")
-                set(OPENSSL_CFLAGS_OTHER "")
-                set(OPENSSL_LDFLAGS "-L${BREW_OPENSSL_PREFIX}/lib;-lssl;-lcrypto")
-        else()
-            message(FATAL_ERROR "OpenSSL (or LibreSSL) is required for building Netdata, but could not be found.")
+                        add_library(PkgConfig::CRYPTO IMPORTED)
+                        set_target_properties(PkgConfig::CRYPTO
+                                IMPORTED_LOCATION ${BREW_OPENSSL_PREFIX}/lib/libcrypto.dylib
+                                INTERFACE_INCLUDE_DIRECTORIES ${BREW_OPENSSL_PREFIX}/include)
+
+                        add_library(PkgConfig::TLS IMPORTED)
+                        set_target_properties(PkgConfig::TLS
+                                IMPORTED_LOCATION ${BREW_OPENSSL_PREFIX}/lib/libssl.dylib
+                                INTERFACE_LINK_LIBRARIES PkgConfig::CRYPTO
+                                INTERFACE_INCLUDE_DIRECTORIES ${BREW_OPENSSL_PREFIX}/include)
+                else()
+                    message(FATAL_ERROR "OpenSSL (or LibreSSL) is required for building Netdata, but could not be found.")
```

Comment:
Hello @Ferroin ,

I am having this message during compilation time:

```sh
CMake Error at CMakeLists.txt:596 (message):
  OpenSSL (or LibreSSL) is required for building Netdata, but could not be
  found.
````

but, I have OpenSSL:

```
-- Checking for module 'openssl'
--   Found openssl, version 3.3.0
-- Configuring incomplete, errors occurred!
````

I also ran the following command to help you:

```sh
# openssl --version
OpenSSL 3.3.0 9 Apr 2024 (Library: OpenSSL 3.3.0 9 Apr 2024)
```

Reply:
I tried to compile on Linux.

Reply:
Should be fixed now.

Reply:
The issue was fixed, please, rebase your PR for me retest it on Linux and also on FreeBSD.